include/fc/io/raw.hpp

```diff
@@ -89,52 +90,66 @@ namespace fc {
     template<typename Stream>
     inline void pack( Stream& s, const fc::time_point_sec& tp, uint32_t _max_depth )
     {
-       uint32_t usec = tp.sec_since_epoch();
-       s.write( (const char*)&usec, sizeof(usec) );
+       pack( s, tp.sec_since_epoch(), _max_depth );
     }
 
     template<typename Stream>
     inline void unpack( Stream& s, fc::time_point_sec& tp, uint32_t _max_depth )
     { try {
        uint32_t sec;
-       s.read( (char*)&sec, sizeof(sec) );
+       unpack( s, sec, _max_depth );
        tp = fc::time_point() + fc::seconds(sec);
     } FC_RETHROW_EXCEPTIONS( warn, "" ) }
 
     template<typename Stream>
     inline void pack( Stream& s, const fc::time_point& tp, uint32_t _max_depth )
     {
-       uint64_t usec = tp.time_since_epoch().count();
-       s.write( (const char*)&usec, sizeof(usec) );
+       pack( s, tp.time_since_epoch().count(), _max_depth );
     }
 
     template<typename Stream>
     inline void unpack( Stream& s, fc::time_point& tp, uint32_t _max_depth )
     { try {
        uint64_t usec;
-       s.read( (char*)&usec, sizeof(usec) );
+       unpack( s, usec, _max_depth );
        tp = fc::time_point() + fc::microseconds(usec);
     } FC_RETHROW_EXCEPTIONS( warn, "" ) }
 
     template<typename Stream>
     inline void pack( Stream& s, const fc::microseconds& usec, uint32_t _max_depth )
     {
-       uint64_t usec_as_int64 = usec.count();
-       s.write( (const char*)&usec_as_int64, sizeof(usec_as_int64) );
+       pack( s, usec.count(), _max_depth );
     }
 
     template<typename Stream>
     inline void unpack( Stream& s, fc::microseconds& usec, uint32_t _max_depth )
     { try {
        uint64_t usec_as_int64;
-       s.read( (char*)&usec_as_int64, sizeof(usec_as_int64) );
+       unpack( s, usec_as_int64, _max_depth );
        usec = fc::microseconds(usec_as_int64);
     } FC_RETHROW_EXCEPTIONS( warn, "" ) }
 
     template<typename Stream, typename T, size_t N>
-    inline void pack( Stream& s, const fc::array<T,N>& v, uint32_t _max_depth ) {
-      s.write((const char*)&v.data[0],N*sizeof(T));
+    inline void pack( Stream& s, const fc::array<T,N>& v, uint32_t _max_depth ) = delete;
```

Comment:
I'm getting a build error on this:

    ../libraries/fc/include/fc/io/raw.hpp:133:83: error: deleted definition must be first declaration
        inline void pack( Stream& s, const fc::array<T,N>& v, uint32_t _max_depth ) = delete;
                                                                                      ^
    ../libraries/fc/include/fc/io/raw_fwd.hpp:131:65: note: previous declaration is here
        template<typename Stream, typename T, size_t N> inline void pack( Stream& s, const fc::array<T,N>& v, uint32_t _max_depth=FC_PACK_MAX_DEPTH );
                                                                    ^


And similar on `unpack` below:

    ../libraries/fc/include/fc/io/raw.hpp:144:79: error: deleted definition must be first declaration
        inline void unpack( Stream& s, fc::array<T,N>& v, uint32_t _max_depth ) = delete;
                                                                                  ^
    ../libraries/fc/include/fc/io/raw_fwd.hpp:132:65: note: previous declaration is here
        template<typename Stream, typename T, size_t N> inline void unpack( Stream& s, fc::array<T,N>& v, uint32_t _max_depth=FC_PACK_MAX_DEPTH);
                                                                    ^

Reply:
Seems the solution is simply to delete the lines in `raw.hpp` and put the `= delete;` on the forward declarations in `raw_fwd.hpp`?

Reply:
Done.